### ISSUE TITLE
Allow to generate a new Auth token

### ISSUE THREAD

**Issue by deruelle** [RestComm/Restcomm-Connect#1386]
It's currently impossible through the Admin UI to generate a new auth token if there was a security breach.

**Comment by deruelle** [comment]
This is true for the REST API as well

**Comment by otsakir** [comment]
@deruelle ,you mean besides updating an account with a new password ?
https://github.com/RestComm/RestComm-Connect/blob/d0204408c22c8376fd464d67cfb402bd447de2a9/restcomm/restcomm.http/src/main/java/org/mobicents/servlet/restcomm/http/AccountsEndpoint.java#L306-L306

**Comment by deruelle** [comment]
No I mean updating the "Auth Token" that allows you to perform REST API calls to your account.

**Comment by otsakir** [comment]
@deruelle , i don't understand. There is currently only one credential field in the Account entity. That's AuthToken. The actual value of AuthToken is rendered from the password when the user creates/updates an account. In that sense, AuthToken is updatable since password is.

**Comment by deruelle** [comment]
mmm actually I think Auth Token should be independent from account password and also we should allow to create more than one to avoid downtime in case you need to change it: 

See https://support.twilio.com/hc/en-us/articles/223136027-Auth-Tokens-and-how-to-change-them
similar is possible with nexmo as well

**Comment by otsakir** [comment]
@deruelle, i totally agree with having two types of credentials. A password and an AuthToken. We use the password to login. We use the AuthToken to access the APIs.

Note that:

1. Twilio asks for the **password** when requesting a new AuthToken.
2. Twilio only allows changing the FriendlyName and Status when updating an account. It does not allow changing the AuthToken/password.

I made some thoughts on refactoring, still immature:

* Alter Account entity. Let it have 'password', 'AuthToken' fields.
* Account creation will be done normally by supplying a 'password' param. The semantics changes here. *password* will be hashed and stored in the 'password' field.  AuthToken, however is initialized using a random value.
* Using the *password* as credential for accessing typical REST API operations, won't be supported. That's what the AuthToken is for. 
* One can use the *password* to retrieve an account. Then, one uses the AuthToken in the account to do typical API operations.

Password/AuthToken reset operations and how we treat sub-account need more thinking. Here are some initial thoughts:

* A separate API path should be used for reseting AuthTokens. It could be hosted under Accounts/Access/ or sth like that or someplace else. This path will authenticate against the actual 'password' only. 
* What about reseting the actual 'password' ???
* What sort of access will parent accounts have over their sub-accounts' 'password' and 'AuthToken' fields. Will they be allowed to change them? Using what sort of credentials and which API endpoint ?

**Comment by deruelle** [comment]
@otsakir your thoughts on refactoring seems correct to me.

resetting the actual password is currently done through the Rest API using the AuthToken.

> What sort of access will parent accounts have over their sub-accounts' 'password' and 'AuthToken' fields
I think they should be allowed to changed them indeed in case a hacker tamper with the sub accounts. Same thing for the Super Admin account using their own credentials by specifying the Account Sid of the account to change the AuthToken or password.

Also I think we need to allow multiple AuthToken at least 2 per account. So that if an AuthToken needs to be changed but a number of applications depends on the initial one, it would allow to transition apps depending on the initial smoothly to a new AuthToken and not break because the AuthToken was changed. Once all apps of the user depending on the AuthToken are changed the initial AuthToken can be removed
